### PR TITLE
ARROW-14761: [Doc] Helping with documentation

### PR DESCRIPTION
**Pull request by AlenkaF**
Add **Helping with documentation** section of the New Contributor's guide.

### CODE REVIEW

File: docs/source/developers/guide/documentation.rst [hunk 1]
```diff
@@ -27,4 +27,65 @@
...
+on JIRA.
+
+.. note::
+	When searching for JIRA issue that deals with documentation
+	try selecting **Components** from **More** tab in JIRA search
+	and there find **Documentation** from the list.
+
+	See `Example search. <https://issues.apache.org/jira/browse/ARROW-14280?jql=project%20%3D%20ARROW%20AND%20status%20%3D%20Open%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20%3D%20Documentation%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC>`_
+
+.. figure:: jira_doc_1.jpeg
```
Comment (pitrou) [inline docs/source/developers/guide/documentation.rst:53]: Should these figures perhaps be indented so as to be displayed inside the note?

File: docs/source/developers/guide/documentation.rst [hunk 2]
```diff
@@ -27,4 +27,65 @@
...
+for corrections is not a good way to go. In this case you can
+search for an issue that is dealing with the documentation
+on JIRA.
+
+.. note::
+	When searching for JIRA issue that deals with documentation
+	try selecting **Components** from **More** tab in JIRA search
+	and there find **Documentation** from the list.
+
+	See `Example search. <https://issues.apache.org/jira/browse/ARROW-14280?jql=project%20%3D%20ARROW%20AND%20status%20%3D%20Open%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20%3D%20Documentation%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC>`_
```
Comment (pitrou) [inline docs/source/developers/guide/documentation.rst:51]: This URL points to a specific issue, we should give the link to the search summary instead, e.g. https://issues.apache.org/jira/issues/?jql=project%20%3D%20ARROW%20AND%20status%20%3D%20Open%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20%3D%20Documentation%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC

File: docs/source/developers/guide/documentation.rst [hunk 3]
```diff
@@ -27,4 +27,65 @@
...
+with the rest of the community.
+
+If you haven't come across an error in the documentation
+it is hard to find one and going through the docs in search
+for corrections is not a good way to go. In this case you can
+search for an issue that is dealing with the documentation
+on JIRA.
+
+.. note::
+	When searching for JIRA issue that deals with documentation
```
Comment (pitrou) [inline docs/source/developers/guide/documentation.rst:47]: It seems your editor is configured to use tab characters for indentation, can you change it to only insert space characters?

(also, conventionally, the indentation size for `.rst` files is 3 spaces :-))

File: docs/source/developers/guide/documentation.rst [hunk 4]
```diff
@@ -27,4 +27,65 @@
...
+Documentation improvements are also a great way to gain some
+experience with our submission and review process without
+requiring a lot of local development environment setup. 
+
+.. note::
+	Many documentation-only changes can be made directly in the
+	GitHub web interface by clicking the **edit** icon. This
+	will handle making a fork and a pull request for you.
+
+.. figure:: github_edit_1.jpeg
```
Comment (pitrou) [inline docs/source/developers/guide/documentation.rst:74]: Same suggestion here (put those inside the note).

**Review by pitrou — COMMENTED** [review]
This is great, thank you very much. Just a couple minor suggestions.

File: docs/source/developers/guide/documentation.rst (hunk 3 above)
Comment (AlenkaF) [reply docs/source/developers/guide/documentation.rst:47]: Yeah, I got bitten by this yesterday! It should be correctly set in the new commit.

**Comment by AlenkaF** [comment]
Failing build doesn't seem to be related (RStudio Package Manager error).

File: docs/source/developers/guide/documentation.rst [hunk 5]
```diff
@@ -27,4 +27,65 @@
...
+A great way to contribute to the project is to improve
+documentation. 
+
+If you are an Arrow user and you found some docs to be
+incomplete or inaccurate, share your hard-earned knowledge
+with the rest of the community.
+
+If you haven't come across an error in the documentation
+it is hard to find one and going through the docs in search
+for corrections is not a good way to go. In this case you can
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/documentation.rst:42]: This first sentence is a bit hard to follow. Maybe adding a comma after the first line would already help. 

Maybe just saying "If you didn't come across something to improve in the documentation itself, you can search for an issue ..." is sufficient?

File: docs/source/developers/guide/documentation.rst [hunk 6]
```diff
@@ -27,4 +27,65 @@
...
+search for an issue that is dealing with the documentation
+on JIRA.
+
+.. note::
+   When searching for JIRA issue that deals with documentation
+   try selecting **Components** from **More** tab in JIRA search
+   and there find **Documentation** from the list.
+
+   See `Example search. <https://issues.apache.org/jira/issues/jql=project%20%3D%20ARROW%20AND%20status%20%3D%20Open%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20%3D%20Documentation%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC>`_
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/documentation.rst:51]:
```suggestion
   See `Example search. <https://issues.apache.org/jira/issues/?jql=project%20%3D%20ARROW%20AND%20status%20%3D%20Open%20AND%20resolution%20%3D%20Unresolved%20AND%20component%20%3D%20Documentation%20AND%20assignee%20in%20(EMPTY)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC>`_
```

(otherwise the link doesn't work for me)

**Review by jorisvandenbossche — COMMENTED** [review]
This is a nice intro for guiding people to fix doc issues!

Not necessarily for this PR (and maybe also for the general doc building page), but while reading through this, I was thinking it would also be nice to explain somewhere:
- Where are the documentation sources located (i.e. if you are on a page on the online docs, which file to edit?)
- The difference between the prose docs (located in /docs) and the docstrings / doc comments (which also again relates to where to go looking for the source to edit)

**Comment by jorisvandenbossche** [comment]
> Where are the documentation sources located (i.e. if you are on a page on the online docs, which file to edit?)

For the prose docs, the theme we are using actually has a "edit this page on github" button that can be enabled (https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/configuring.html#add-an-edit-this-page-button)

File: docs/source/developers/guide/documentation.rst (hunk 5 above)
Comment (AlenkaF) [reply docs/source/developers/guide/documentation.rst:42]: Yes, you are quite right. It's a confusing sentence. Will correct, thanks!

**Comment by AlenkaF** [comment]
> * Where are the documentation sources located (i.e. if you are on a page on the online docs, which file to edit?)

That would be great, yes! I will add it into this file, I think it fits nicely.

> * The difference between the prose docs (located in /docs) and the docstrings / doc comments (which also again relates to where to go looking for the source to edit)

This part is maybe a little more complicated, but would definitely need to be added in the existing docs. I can make a separate issue for it?

> For the prose docs, the theme we are using actually has a "edit this page on github" button that can be enabled (https://pydata-sphinx-theme.readthedocs.io/en/latest/user_guide/configuring.html#add-an-edit-this-page-button), that might be interesting to add and show here in combination with the screenshots about how to edit on github (the button itself only brings you to that page on github)

Also something we already talked about! Big +1, will add it to `conf.py` and add the description into this PR.

File: docs/source/developers/guide/documentation.rst [hunk 7]
```diff
@@ -27,4 +27,65 @@
...
+
+If you are an Arrow user and you found some docs to be
+incomplete or inaccurate, share your hard-earned knowledge
+with the rest of the community.
+
+If you haven't come across an error in the documentation
+it is hard to find one and going through the docs in search
+for corrections is not a good way to go. In this case you can
+search for an issue that is dealing with the documentation
+on JIRA.
```
Comment (pitrou) [inline docs/source/developers/guide/documentation.rst:44]: Perhaps add something about the "good-first-issue" and "good-second-issue" labels? Example search:
https://issues.apache.org/jira/issues/?jql=project%20%3D%20ARROW%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20in%20(good-first-issue%2C%20good-second-issue)%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC

Comment (pitrou) [reply docs/source/developers/guide/documentation.rst:44]: Oh, actually it's done in https://github.com/apache/arrow/pull/11807 . Nevermind.

**Comment by AlenkaF** [comment]
@jorisvandenbossche @pitrou I added the suggestions, Edit this Page button and source locations for the docs. Think this is ready for another round of review.

File: docs/source/developers/guide/documentation.rst [hunk 8]
```diff
@@ -27,4 +27,79 @@
...
+incomplete or inaccurate, share your hard-earned knowledge
+with the rest of the community.
+
+If you didn't come across something to improve in the
+documentation itself, you can search for an issue on JIRA.
+
+.. note::
+   When searching for JIRA issue that deals with documentation
+   try selecting **Components** from **More** tab in JIRA search
+   and there find **Documentation** from the list.
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/documentation.rst:45]:
```suggestion
   When searching for JIRA issue that deals with documentation,
   try selecting **Components** from the **More** tab in JIRA search
   and select **Documentation** from the list.
```

File: docs/source/developers/guide/documentation.rst [hunk 9]
```diff
@@ -27,4 +27,79 @@
...
+      :scale: 30 %
+      :alt: edit file in GitHub.
+
+      Now you can edit the file in GitHub.
+
+You could also build the entire project, make the change locally on
+your branch and make the PR this way. But it is by no means superior
+to simply editing via GitHub.
+
+If you wish to build the documentation also, follow detailed instructions
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/documentation.rst:87]:
```suggestion
If you wish to build the documentation locally, follow detailed instructions
```

File: docs/source/developers/guide/documentation.rst [hunk 10]
```diff
@@ -27,4 +27,79 @@
...
+
+Most of the documentation is located in the ``docs/source`` of the Arrow
+library. Source folder includes:
+
+- **C++ documentation** section: ``docs/source/cpp``.
+- **Development** section: ``docs/source/developers``.
+- **Specificatons and protocols** section: ``docs/source/format``.
+- Other language documentation: **C (GLib), Java, JavaScript** and **Python**
+  in folders named by the language, for example: ``docs/source/c_glib``.
+- The documentation for the **R language** is located in the ``r/sub-directory``.
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/documentation.rst:101]:
```suggestion
- The documentation for the **R language** is located in the ``r/`` sub-directory.
```

?

File: docs/source/developers/guide/documentation.rst [hunk 11]
```diff
@@ -27,4 +27,79 @@
...
+to simply editing via GitHub.
+
+If you wish to build the documentation also, follow detailed instructions
+on :ref:`building-docs`.
+
+Where to find the correct file to change?
+-----------------------------------------
+
+Most of the documentation is located in the ``docs/source`` of the Arrow
+library. Source folder includes:
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/documentation.rst:94]: Now you added this list, I think it is relevant to mention "docstrings" for Python as well -> for editing the reference docs, you will need to edit the docstring the pyarrow package itself

Comment (jorisvandenbossche) [reply docs/source/developers/guide/documentation.rst:94]: And actually for C++ it's the same issue

Comment (AlenkaF) [reply docs/source/developers/guide/documentation.rst:94]: Ok, will do.

File: docs/source/developers/guide/documentation.rst [hunk 12]
```diff
@@ -27,4 +27,95 @@
...
+
+- **C++ documentation** section: ``docs/source/cpp``.
+- **Development** section: ``docs/source/developers``.
+- **Specificatons and protocols** section: ``docs/source/format``.
+- Other language documentation: **C (GLib), Java, JavaScript** and **Python**
+  in folders named by the language, for example: ``docs/source/c_glib``.
+- The documentation for the **R language** is located in the ``r/`` sub-directory.
```
Comment (AlenkaF) [inline docs/source/developers/guide/documentation.rst:101]: @thisisnic did I get this R part of the documentation source right?

File: docs/source/developers/guide/documentation.rst [hunk 13]
```diff
@@ -27,4 +27,95 @@
...
+  - In the ``R/vignettes`` you can find the **Articles**, for example
+    `Working with Cloud Storage (S3) <https://arrow.apache.org/docs/r/articles/fs.html>`_.
+  - In the ``r/R`` you can edit reference documentation for example 
+    in the ``r/R/dataset.R`` you can find the ``docstrings`` for the
+    `open_dataset <https://arrow.apache.org/docs/r/reference/open_dataset.html>`_.
+
+- **Reference documentation**
+
+  To edit `API reference documentation for Python <https://arrow.apache.org/docs/python/api.html>`_,
+  you will need to edit the ``docstring`` of the ``pyarrow`` package itself.
```
Comment (jorisvandenbossche) [inline docs/source/developers/guide/documentation.rst:112]:
```suggestion
  you will need to edit the docstrings of the functions and methods located in the ``pyarrow`` package itself.
```

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!
